src/storage/core.rs

```diff
@@ -897,30 +901,33 @@ where
     async fn mark_all_as_deleted_closed(&self, key: &K) -> Result<u64> {
         let safe = self.inner.safe.write().await;
         let mut blobs = safe.blobs.write().await;
-        let entries_closed_blobs = blobs
-            .iter_mut()
-            .map(|b| b.mark_all_as_deleted(key))
-            .collect::<FuturesUnordered<_>>();
-        let total = entries_closed_blobs
-            .filter_map(|result| match result {
-                Ok(count) => count,
-                Err(error) => {
-                    warn!("failed to delete records: {}", error);
-                    None
+        let count = AtomicU64::new(0);
+        blobs
+            .for_each(|b| async {
+                let rwlock = ASRwLock::new(b);
```

Comment:
Oh, it is not good to wrap blob with RwLock only in order to use the new signature of `Blob::mark_all_as_deleted`. I think, it is better just to revert to the original implementation, where `mark_all_as_deleted` just received `Blob` as `self`